internal/upgrade/pkg.go

```diff
@@ -0,0 +1,91 @@
+package upgrade
+
+import (
+	"strings"
+
+	"github.com/talos-systems/bldr/internal/pkg/types/v1alpha1"
+	"github.com/talos-systems/bldr/internal/pkg/types/v1alpha2"
+)
+
+// FromV1Alpha1 upgrades v1alpha1 format -> v1alpha2
+func FromV1Alpha1(oldPkg *v1alpha1.Pkg, stageNames []string) *v1alpha2.Pkg {
+	convertDeps := func(old []*v1alpha1.Dependency) v1alpha2.Dependencies {
+		new := v1alpha2.Dependencies{}
+		for _, dep := range old {
+			src := dep.Image
+			src = strings.SplitN(src, ":", 2)[0]
```

Comment:
since we're overwriting src here anyways, does it make sense to just use `dep.Image` for `strings.SplitN` ?

Reply:
nice catch!